app/os_windows.go

```diff
@@ -954,4 +1001,66 @@ func configForDPI(dpi int) unit.Metric {
 	}
 }
 
+// schemesURI is a list of schemes, comma separated, that must be
+// defined using -X compiler ldflag, that used in gogio.
+var schemesURI string
+
+// startupURI is the URI that started the app, if any.
+var startupURI string
+
+func init() {
+	var currentSchemes []string
+	if schemesURI != "" {
+		currentSchemes = strings.Split(schemesURI, ",")
+	}
+
+	oldSchemes := windows.RegisteredSchemes(appID)
+	for _, s := range currentSchemes {
+		for i, o := range oldSchemes {
+			if s == o {
+				oldSchemes = append(oldSchemes[:i], oldSchemes[i+1:]...)
+				break
+			}
+		}
+	}
+
+	if len(oldSchemes) > 0 {
+		go windows.UnregisterSchemes(appID, oldSchemes)
+	}
+
+	if len(currentSchemes) == 0 {
+		return
+	}
+
+	if len(os.Args) >= 3 && os.Args[1] == "-gio_launch_url" {
```

Comment:
Use equality, `len(os.Args) == 3`, so there's no doubt about Windows launching us, not the user.